Add support for using Eclipse Zenoh as transport between vehicle and back-end

vivekpandey02 opened the pull request:
I am pleased to present a Pull Request marking the initiation of the Eclipse Zenoh integration into the Eclipse SDV FMS blueprint. This PR focuses on extending the FMS Forwarder to publish data via Zenoh, alongside configuring the Zenoh Router to receive data over MQTT. This integration sets the stage for the subsequent phase, where FMS consumer integration into the Zenoh Router will be addressed, further strengthening the capabilities of our Fleet Management System Blueprint. Please review the PR and provide the necessary feedback.

vivekpandey02 committed:
updated zenoh code

vivekpandey02 committed:
Revert "updated zenoh code"

This reverts commit 1009a8c9cb19c312aff7013043e7960f7db6b07d.

vivekpandey02 committed:
Added code to support zenoh communication

vivekpandey02 committed:
Added the zenoh support in Cargo.toml & readme.

vivekpandey02 committed:
Added the yaml file for zenoh router.

vivekpandey02 committed:
added the fms zenoh communication architecture diagram.

vivekpandey02 committed:
add the zenoh-config.json file for the zenoh router.

vivekpandey02 committed:
Updated the readme file for zenoh communication.

vivekpandey02 committed:
Update README.md

vivekpandey02 committed:
Updated the Cargo.lock file to Downgrading duration-str v0.7.1 -> v0.7.0

vivekpandey02 committed:
Update and rename zenoh-config.json to zenoh-config.json5

vivekpandey02 committed:
Update fms-blueprint-compose-zenoh.yaml

vivekpandey02 committed:
Update zenoh-config.json5

vivekpandey02 committed:
Update zenoh-config.json5

vivekpandey02 committed:
Update zenoh-config.json5

vivekpandey02 committed:
Update fms-blueprint-compose-zenoh.yaml

pgangula-src committed:
Changes in README.md

TIny changes in readme to provide information on the test application to validate the forwarder data.

eriksven commented:
Hello,

thank you for the PR. 

Regarding the failing the "Lint source code / lint-source-code" Check: We just created a PR in #37, which should fix this issue. 

Regarding the ECA-Check: When I click on the details the same address is listed as covered once and as not covered 14 times. Could it be that. you signed the ECA after making the respective commits?

sophokles73 commented on components/fms-forwarder/src/zenoh_publisher.rs:
```diff
@@ -0,0 +1,115 @@
+                .value_parser(clap::value_parser!(WhatAmI))
+                .long("mode")
+                .short('m')
+                .help("-m, --mode=[MODE] 'The zenoh session mode (peer by default).")
```
there is no need to specify the switches, that will be done by clap. The plain description is sufficient.

vivekpandey02 replied:
yes, modified it.

sophokles73 commented on components/fms-forwarder/src/zenoh_publisher.rs:
```diff
@@ -0,0 +1,115 @@
+use protobuf::Message;
+use std::sync::Arc;
+use zenoh::config::Config;
+use zenoh::prelude::sync::*;
```
do we really need this wild card import? Can't we use explicit imports for what we actually use, or are we using almost everything of that module?

vivekpandey02 replied:
We utilize nearly all components from this module; thus, opting for a wildcard import could be a more efficient approach.

sophokles73 commented on fms-blueprint-compose.yaml:
```diff
@@ -145,7 +145,7 @@ services:
     command: "--insecure"
   fms-forwarder:
-    image: "ghcr.io/eclipse-sdv-blueprints/fleet-management/fms-forwarder:main"
+    image: "fms-forwarder:custom"
```
this seems to have been changed by mistake?

evshary replied:
I think this is for testing. Will the image include the Zenoh function if merging the PR?
Or do we need to have a way to build a customized image for the Zenoh function for the time being?

sophokles73 replied:
well, my understanding is that no harm will be done by adding the `zenoh` subcommand, right? There is currently no point in using it, but it doesn't break any of the existing functionality. So, FMPOV we could simply keep this as an *undocumented* feature for the time being. WDYT?

pgangula-src replied:
Just like in the Hono case, we'll apply the logic in the FMS Consumer to receive data coming from Zenoh over the Zenoh Router, store it in InfluxDB, and ensure the project is immediately usable. Also make sure the documentation reflects that. WDYT?

sophokles73 commented on fms-blueprint-compose-zenoh.yaml:
```diff
@@ -0,0 +1,17 @@
+      RUST_LOG: zenoh=info
+    image: eclipse/zenoh:0.10.1-rc
+    container_name: "fms-zenoh-router"
+    network_mode: host
```
this doesn't seem right. In fact, I would have expected to find:
```yaml
networks:
- "fms-vehicle"
- "fms-backend"
```

instead ...

vivekpandey02 replied:
yes, modified it.

sophokles73 commented on README.md:
````diff
@@ -99,6 +99,39 @@ of an Eclipse Hono instance as shown in the diagram below.
    back end component which receives vehicle data via Hono's north bound Kafka based Telemetry API and writes it to the
    Influx DB.
 
+
+# Using Zenoh Router to geographically distribute the vehicle data
+
+The integration of Eclipse Zenoh into the FMS blueprint is planned to be realized in two steps:
+
+**Step 1:  Extend FMS Forwarder to publish data in Zenoh and configure Zenoh Router to also receive data over MQTT**
+
+- Assess the current FMS Forwarder and identify the necessary modifications to enable data publication through Zenoh.
+- Implement the required code changes to publish data via Zenoh.
+- MQTT plugin configuration on the Zenoh Router to as well support MQTT interfaces.
+
+**Step 2: FMS consumer Integration in Zenoh Router**
+
+- Implement the necessary changes in Zenoh Router using Zenoh Flow to create components that publish / consume data into influxdb.
+
+Currently, step 1 has been implemented, and we now provide the instructions on how you can test the communication using Zenoh, and we are actively advancing with the ongoing work on step 2.
+
+<img src="img/architecture-zenoh.drawio.png">
+
+1. Create a local docker image for FMS Forworder to support the zenoh communication, run the below command from  fleet-management/components/ directory:
+   
+   ```sh
+   docker build -t fms-forwarder:custom -f Dockerfile.fms-forwarder .
+   ```
+   
+2. Start up the zenoh router and back end services using Docker Compose files in the top level (fleet-management/) folder:
+
+   ```sh
+   docker compose -f ./fms-blueprint-compose.yaml -f ./fms-blueprint-compose-zenoh.yaml up --detach
+   ```
+
+Here you find the [Zenoh based Python Subscriber](https://github.com/vivekpandey02/fms-zenoh-test-subscriber) application to validate the data transmissted by FMS Fowarder.
+
````
please only describe the status quo established by this PR. If you extend/amend the functionality in subsequent PRs, you can amend/change the text accordingly.

vivekpandey02 replied:
Yes, modified it.

sophokles73 reviewed (changes requested):
Thanks for the PR :-)
Am I right in assuming that the Zenoh router will currently not forward data to influx DB?

eriksven commented:
In the meantime we merged #37, so to resolve the linting error you can rebase to the current main-branch. 

The issue with the ECA tool seems to have resolved itself in the meantime.

evshary commented:
> Thanks for the PR :-) Am I right in assuming that the Zenoh router will currently not forward data to influx DB?

Not yet. I believe this is the next step. Maybe will use the Zenoh router with the influxdb plugin to forward data.

vivekpandey02 committed:
changes done according to the comments provided in PR #36

vivekpandey02 committed:
changes done according to the comments provided in PR #36

vivekpandey02 committed:
Update README.md

vivekpandey02 committed:
Updated the cargo.lock file for Zenoh.

sophokles73 commented on README.md:
```diff
@@ -99,6 +99,34 @@ of an Eclipse Hono instance as shown in the diagram below.
+- Implement the required code changes to publish data via Zenoh.
+- MQTT plugin configuration on the Zenoh Router to as well support MQTT interfaces.
+
+<img src="img/architecture-zenoh.drawio.png">
```
the architecture illustrated here is not implemented by this PR, or is it? If not, then please either remove the picture altogether or provide a picture that reflects what has been implemented. Otherwise, users might get confused as to what expect ...

vivekpandey02 replied:
We have updated the architecture diagram and our implementation is align with architecture diagram.

sophokles73 commented on README.md:
```diff
@@ -99,6 +99,34 @@ of an Eclipse Hono instance as shown in the diagram below.
+
+- Assess the current FMS Forwarder and identify the necessary modifications to enable data publication through Zenoh.
+- Implement the required code changes to publish data via Zenoh.
+- MQTT plugin configuration on the Zenoh Router to as well support MQTT interfaces.
```
Is there a way to actually make use of this, e.g. using the FMS Forwarder's *hono* command but instead of publishing to a real Hono instance, publishing to zenoh using MQTT?

vivekpandey02 replied:
We can use the FMS Forwarder's hono command instead of publishing to a real Hono instance, publishing to zenoh using MQTT. We need to configure Zenoh MQTT Plugin in zenoh router.

sophokles73 commented on README.md:
````diff
@@ -99,6 +99,34 @@ of an Eclipse Hono instance as shown in the diagram below.
+   docker compose -f ./fms-blueprint-compose.yaml -f ./fms-blueprint-compose-zenoh.yaml up --detach
+   ```
+
+Here you find the [Zenoh based Python Subscriber](https://github.com/vivekpandey02/fms-zenoh-test-subscriber) application to validate the data transmissted by FMS Fowarder.
````
Can't we simply include the necessary code from the other repo in this PR as well?

vivekpandey02 replied:
Readme updated for as per latest zenoh communication architectured.

sophokles73 commented on components/fms-forwarder/Cargo.toml:
```diff
@@ -30,6 +30,7 @@ readme.workspace = true
 
 [dependencies]
 async-trait = { workspace = true }
+async-std = {features = ["attributes"] }
```
please add a (minor) *version* property ...

vivekpandey02 replied:
version added

sophokles73 commented on components/fms-forwarder/Cargo.toml:
```diff
@@ -38,6 +39,7 @@ fms-proto = { workspace = true }
 influx-client = { workspace = true, features = ["writer"] }
 log = { workspace = true }
 paho-mqtt = { version = "0.12", default-features = false, features = [ "vendored-ssl" ] }
+zenoh = {version = "0.10.1-rc"}
```
will this only work with this particular `rc` version or can we also specify e.g. `0.10` as the requirement?

vivekpandey02 replied:
For the time being it works with particular rc version.

sophokles73 commented on components/fms-forwarder/src/zenoh_publisher.rs:
```diff
@@ -0,0 +1,115 @@
+                .required(false),
+        )
+        .arg(
+            Arg::new("no-multicast-scouting")
```
I assume that *multicast scouting* is enabled by default?
If so, then FMPOV this arg should be called `disable-multicast-scouting` and not require a value but instead use a `clap::ArgAction::SetFalse` to set the property to false.

Alternatively, it should be called `use-multicast-scouting` and use a `clap::builder::BoolishValueParser` with a default value of `true` ...

vivekpandey02 replied:
Yes, By default zenoh router replies to multicast scouting messages for being discovered by peers and clients.This option disables this feature. This configuration name is defined in zenoh config api.

sophokles73 replied:
so we are talking about the first option, keeping the long-name. Then please change the `value_parser` accordingly ...

vivekpandey02 replied:
Code updated as per your suggestion.

vivekpandey02 committed:
Added code to support zenoh communication

vivekpandey02 committed:
Updated Cargo.toml for zenoh

vivekpandey02 committed:
Updated the fms-blueprint-compose-zenoh.yaml for fms-consumer

pgangula-src committed:
Update README.md

Latest update with FMS Consumer

sophokles73 commented:
@vivekpandey02 are you still working on this?

vivekpandey02 commented:
> @vivekpandey02 are you still working on this?

Yes, I am working on it.  i have changed the FMS-Consumer for zenoh support. Now FMS Consumer can take the data from Hono Kafka and Zenoh both and write into influxDB. so we introduced Zenoh in FMS Forworder, and FMS-Consumer Both. I will merge my code into my PR#36 soon

vivekpandey02 committed:
Updated Zenoh Communication Architecture Diagram

vivekpandey02 committed:
Update README.md 

Update README for the zenoh communication .

vivekpandey02 committed:
Added code to support zenoh configuration file

vivekpandey02 committed:
added code to support zenoh configuration file.

vivekpandey02 committed:
added readme file for fms-consumer

vivekpandey02 committed:
updated the hono-kafka to kafka in code

vivekpandey02 committed:
Update fms-blueprint-compose-hono.yaml for fms-condumer sub command.

vivekpandey02 committed:
Update fms-blueprint-compose-zenoh.yaml

vivekpandey02 committed:
Update README.md

vivekpandey02 committed:
formated the main file.

vivekpandey02 committed:
Update main.rs

vivekpandey02 committed:
Update README.md

vivekpandey02 committed:
Updated the yaml files

vivekpandey02 committed:
Added the zenoh in component cargo.toml

vivekpandey02 committed:
updated the fms-consumer cargo.toml file

vivekpandey02 committed:
Update the fms-forwarder cargo.toml file

vivekpandey02 committed:
Delete img/architecture-zenoh.drawio.png

vivekpandey02 committed:
Rename run_async_processor() to run_async_processor_kafka()

vivekpandey02 committed:
Merge branch 'fms-forwarder-zenoh-communication' of https://github.com/vivekpandey02/fleet-management into fms-forwarder-zenoh-communication

vivekpandey02 committed:
changes cargo files for async-std version

vivekpandey02 commented:
> @vivekpandey02 are you still working on this?

Hi @sophokles73, I have merged all the changes for Zenoh end to end communication in #PR36

vivekpandey02 commented:
> > @vivekpandey02 are you still working on this?
> 
> Hi @sophokles73, I have merged all the changes for Zenoh end to end communication in #PR36

Hi @sophokles73, is there any comment / Feedback on PR #36 ?

sophokles73 commented on README.md:
```diff
@@ -98,6 +98,25 @@ of an Eclipse Hono instance as shown in the diagram below.
+[Eclipse Zenoh](https://zenoh.io/) offers an exceptionally efficient wire protocol, lightweight discovery, and adaptive routing algorithms that prove highly responsive in dynamic environments and topology changes.
+These results in seamless session migration and ensures optimal latency and resource consumption for devices in decentralized systems, even in the face of mobility. 
+
+Deploying the Fleet Management system over the Zenoh Router allows consumers to access data through the closest path, promoting significant decentralization and improved mobility that are critical to managing a Fleet.
```
I understand that you want to point people to the advantages of Zenoh. However, I would prefer to stick a little more to the facts that are relevant in this context. Please rephrase this in a way that does not sound like its coming from your marketing department ;-)

vivekpandey02 replied:
Read-Me changed accordingly.

sophokles73 commented on components/Cargo.toml:
```diff
@@ -36,6 +36,7 @@ documentation = "https://eclipse-sdv-blueprints.github.io/blueprints-website/"
 
 [workspace.dependencies]
 async-trait = { version = "0.1" }
+async-std = {version = "1.12.0", default-features = false }
```
can we also set the minor version as required, i.e. use `"1.12"`? Can you also add a white space before `version` (just for the looks ;-))

vivekpandey02 replied:
version updated

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -251,13 +352,21 @@ pub async fn main() {
+
+    parser = parser
+        .subcommand_required(true)
+        .subcommand(influx_client::connection::add_command_line_args(
```
FMPOV we do not need to add the influx client config args at the command level again, given that we already define them at the top level.

vivekpandey02 replied:
Code changed accordingly.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -272,10 +381,70 @@ pub async fn main() {
-        );
+        ),
+        ))
+        .subcommand(influx_client::connection::add_command_line_args(
```
see above

vivekpandey02 replied:
Code changed accordingly.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -36,13 +36,58 @@ use rdkafka::consumer::Consumer;
 const PARAM_KAFKA_PROPERTIES_FILE: &str = "kafka-properties-file";
 const PARAM_KAFKA_TOPIC_NAME: &str = "kafka-topic";
 
+const SUBCOMMAND_KAFKA: &str = "kafka";
```
while in general, this could work with any Kafka broker, we rely on the messages being available on a particular topic and expect to find information in message headers defined by Hono. So I would prefer to call the command `hono` and use `HONO` in all places were you use `KAFKA` in constants etc.

vivekpandey02 replied:
Code changed.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -148,7 +193,15 @@ async fn process_protobuf_message(
+    }
+}
+
+async fn process_kafka_message(m: &BorrowedMessage<'_>, influx_writer: Arc<InfluxWriter>) {
```
process_hono_message

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -167,7 +220,7 @@ async fn process_message(m: &BorrowedMessage<'_>, influx_writer: Arc<InfluxWrite
 }
 
-async fn run_async_processor(args: &ArgMatches) {
+async fn run_async_processor_kafka(args: &ArgMatches) {
```
run_async_processor_hono

vivekpandey02 replied:
function name changed accordingly.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -176,12 +229,16 @@ async fn run_async_processor(args: &ArgMatches) {
-                error!("failed to create Kafka client: {e}");
-                process::exit(1);
-            });
+    let kafka_args = args.subcommand_matches(SUBCOMMAND_KAFKA).unwrap();
```
hono_args ...

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -243,6 +302,48 @@ async fn run_async_processor(args: &ArgMatches) {
+
+    let subscriber = session.declare_subscriber(KEY_EXPR).res().await.unwrap();
+
+    println!("Enter 'q' to quit...");
```
can we please instead rely on shutting down via CTRL-C and remove the dependency on `async_std`?

sophokles73 replied:
are you going to change this as well?

vivekpandey02 replied:
Yes, i changed the code as per your comment.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -251,13 +352,21 @@ pub async fn main() {
         .version(version)
-        .about("Receives FMS related VSS data points via Hono's Kafka based Telemetry API and writes them to an InfluxDB server")
-        .arg(
+        .about("Receives FMS related VSS data points via Hono's Kafka based Telemetry API and writes them to an InfluxDB server");
```
we might want to adapt this to either be more generic or to also mention the Zenoh option?

vivekpandey02 replied:
Zenoh Option has been added.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -251,13 +352,21 @@ pub async fn main() {
+        .subcommand_required(true)
+        .subcommand(influx_client::connection::add_command_line_args(
+            Command::new(SUBCOMMAND_KAFKA)
+                .about("Forwards VSS data to an Influx DB server from Kafka").arg(
```
> from Hono's north bound Kafka API

vivekpandey02 replied:
Change done accordingly.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -272,10 +381,70 @@ pub async fn main() {
+        ))
+        .subcommand(influx_client::connection::add_command_line_args(
+            Command::new(SUBCOMMAND_ZENOH)
+                .about("Forwards VSS data to an Influx DB server from zenoh")
```
you seem to use `zenoh` nut also `Zenoh` in different places. Is there an *official* way of spelling it? In any case, I would like to use consistent spelling/naming throughout the code, if possible.

vivekpandey02 replied:
Officialy it is Zenoh. I changed zenoh to Zenoh across the readme and statements.

sophokles73 commented on components/fms-forwarder/README.md:
```diff
@@ -62,3 +62,15 @@ Please refer to the command line help for details:
+## Publishing to Zenoh Router
+
+The forwarder can publish status information to the Zenoh Router of an [Eclipse Zenoh](https://projects.eclipse.org/projects/iot.zenoh/) instance.
+For this to work, the forwarder needs to be configured with the zenoh router end points,
```
zenoh vs Zenoh?

vivekpandey02 replied:
Change done.

sophokles73 commented on components/fms-forwarder/src/main.rs:
```diff
@@ -78,10 +84,20 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             // cannot happen because subcommand is required
             process::exit(1);
-        }
+        },
```
please remove the trailing comma

vivekpandey02 replied:
Change done

sophokles73 commented on fms-blueprint-compose-hono.yaml:
```diff
@@ -19,6 +19,7 @@
 
 services:
   fms-consumer:
+    command: "kafka"
```
hono

vivekpandey02 replied:
changed done accordingly.

sophokles73 commented on components/fms-consumer/README.md:
````diff
@@ -0,0 +1,58 @@
+fms-consumer --help
+```
+
+## Received data from Hono's north bound Kafka based Telemetry API 
````
> Receive data from Hono's ...

vivekpandey02 replied:
changed done accordingly.

sophokles73 commented on components/fms-consumer/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Received data from Hono's north bound Kafka based Telemetry API 
+
+The consumer can received data from Hono's north bound Kafka based Telemetry API 
```
> ... can receive ...

vivekpandey02 replied:
changed done accordingly.

sophokles73 commented on components/fms-consumer/README.md:
````diff
@@ -0,0 +1,58 @@
+```
+
+
+## Received data from Zenoh Router
````
> Receive data ...

vivekpandey02 replied:
changed done accordingly.

sophokles73 commented on components/fms-consumer/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Received data from Zenoh Router
+
+The consumer can subscriber status from Zenoh Router of an [Eclipse Zenoh](https://projects.eclipse.org/projects/iot.zenoh/) instance.
```
> The consumer can receive data from an [Eclipse Zenoh](https://projects.eclipse.org/projects/iot.zenoh/) Router instance.

vivekpandey02 replied:
changed done accordingly.

sophokles73 reviewed (changes requested):
Sorry for the delay and thanks for adding the code for consuming data via Zenoh :+1:

vivekpandey02 committed:
Formatted the fms-forwarder zenoh code

vivekpandey02 committed:
Updated the code in fms-consumer main.

vivekpandey02 committed:
Updated the fms-consumer readme

vivekpandey02 committed:
Update Cargo.toml async-std version (added minor version)

vivekpandey02 committed:
Update README.md for Zenoh

vivekpandey02 committed:
Update fms-blueprint-compose-hono.yaml

change the command from kafka to hono

sophokles73 commented on README.md:
```diff
@@ -98,6 +98,22 @@ of an Eclipse Hono instance as shown in the diagram below.
+
+<img src="img/architecture-zenoh.drawio.svg">
+
+Start up the zenoh router and other services using Docker Compose files in the top level (fleet-management/) folder:
```
Zenoh router?

vivekpandey02 replied:
change done accordingly.

sophokles73 commented on components/fms-consumer/README.md:
````diff
@@ -0,0 +1,57 @@
+Please refer to the command line help for details:
+
+```sh
+fms-consumer kafka --help
````
`fms-consumer hono --help`?

vivekpandey02 replied:
change done accordingly.

vivekpandey02 commented:
Hello @sophokles73, I've incorporated the changes based on your review comments in my pull request.

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -176,12 +229,16 @@ async fn run_async_processor(args: &ArgMatches) {
-                error!("failed to create Kafka client: {e}");
-                process::exit(1);
-            });
+    let kafka_args = args.subcommand_matches(SUBCOMMAND_HONO).unwrap();
```
`hono_args`

vivekpandey02 replied:
changed the name accordingly

sophokles73 commented on components/fms-consumer/src/main.rs:
```diff
@@ -251,13 +352,21 @@ pub async fn main() {
                 .value_parser(clap::builder::NonEmptyStringValueParser::new())
                 .long(PARAM_KAFKA_PROPERTIES_FILE)
-                .help("The path to a file containing Kafka client properties for connecting to the Kafka broker(s).")
+                .help("The path to a file containing Kafka client properties for connecting to the Kafka broker(s).")                
```
can you please remove the trailing whitespace?

vivekpandey02 replied:
trailing whitespace removed

sophokles73 reviewed (changes requested):
Looks pretty good already :+1:
Can you address the few (small) remaining issues? Then we can squash and merge ...

vivekpandey02 committed:
Update README.md for Zenoh

vivekpandey02 committed:
Update README.md

vivekpandey02 committed:
Update fms-consumer README.md

vivekpandey02 committed:
Update fms-consumer/main.rs for hono_args and whitespace

vivekpandey02 committed:
formatted the fms-consumer main.rs

vivekpandey02 committed:
Remove the code for shutting down via CTRL-C and remove the dependency on async_std

vivekpandey02 committed:
updated fms-consumer readme for zenoh

vivekpandey02 committed:
Updated the fms forwarder code for Zenoh

vivekpandey02 committed:
Updated fms-forwarder readme for Zenoh.

vivekpandey02 committed:
Updated the readme for Zenoh

vivekpandey02 committed:
Updated fms-blueprint-compose-zenoh.yaml for zenoh dependency

vivekpandey02 committed:
Updated fms-blueprint-compose-zenoh.yaml for zenoh dependency

sophokles73 commented on components/Cargo.toml:
```diff
@@ -36,6 +36,7 @@ documentation = "https://eclipse-sdv-blueprints.github.io/blueprints-website/"
 
 [workspace.dependencies]
 async-trait = { version = "0.1" }
+async-std = {version = "1.12", default-features = false }
```
do we still need this?

vivekpandey02 replied:
async-std removed from Cargo.toml files.

sophokles73 commented on components/fms-forwarder/Cargo.toml:
```diff
@@ -30,6 +30,7 @@ readme.workspace = true
 
 [dependencies]
 async-trait = { workspace = true }
+async-std = {workspace = true, features = ["attributes"] }
```
do we still need this?

vivekpandey02 replied:
async-std removed from Cargo.toml files.

vivekpandey02 committed:
Remove async-std from Cargo.toml files.

vivekpandey02 commented:
Hi @sophokles73, I have done changes for the Zenoh Code for PR #36 as per your feedback.

vivekpandey02 committed:
Change process_kafka_message() to process_hono_message

vivekpandey02 committed:
Update Cargo.lock

vivekpandey02 committed:
Update Cargo.lock

sophokles73 reviewed (approved):
LGTM

Thanks for contributing and your patience :+1: 

I have seen that you have undone your changes to Cargo.lock, which is fine, I will make the required changes after merging.
Do you want to squash your branch into a reasonable (small) number of commits, or should I do it when merging?

vivekpandey02 commented:
> LGTM
> 
> Thanks for contributing and your patience 👍
> 
> I have seen that you have undone your changes to Cargo.lock, which is fine, I will make the required changes after merging. Do you want to squash your branch into a reasonable (small) number of commits, or should I do it when merging?

You're welcome! I appreciate your guidance and patience throughout the process. I'm glad to contribute to the project. I trust your judgment, so if you could handle the squashing during the merging process, that would be fantastic. I appreciate your cooperation.
